std/algorithm/searching.d

```diff
@@ -799,15 +802,33 @@ if (isForwardRange!R
                   haystack.popFront();
             }
         }
+        //Because of @@@8804@@@: Avoids both "unreachable code" or "no return statement"
+        static if (isInfinite!R) assert(0);
```

Comment:
Why? There are legitimate use cases for using `countUntil` with an infinite range (I added also a test to ensure that this works)